src/coreclr/vm/jitinterface.cpp

```diff
@@ -10859,41 +10846,68 @@ void CEECodeGenInfo::getHelperFtn(CorInfoHelpFunc    ftnNum,               /* IN
                     {
                         // Cache it for future uses to avoid taking the lock again.
                         hlpFinalTierAddrTable[dynamicFtnNum] = finalTierAddr;
-                        if (pNativeEntrypoint != NULL)
-                        {
-                            pNativeEntrypoint->accessType = IAT_VALUE;
-                            pNativeEntrypoint->addr = finalTierAddr;
-                        }
+                        accessType = IAT_VALUE;
+                        targetAddr = finalTierAddr;
                         goto exit;
                     }
                 }
             }
 
             if (IndirectionAllowedForJitHelper(ftnNum))
             {
+#ifdef FEATURE_PORTABLE_ENTRYPOINTS
+                accessType = IAT_VALUE;
+                targetAddr = (LPVOID)helperMD->GetPortableEntryPoint();
+#else // !FEATURE_PORTABLE_ENTRYPOINTS
                 Precode* pPrecode = helperMD->GetPrecode();
                 _ASSERTE(pPrecode->GetType() == PRECODE_FIXUP);
-                if (pNativeEntrypoint != NULL)
-                {
-                    pNativeEntrypoint->accessType = IAT_PVALUE;
-                    pNativeEntrypoint->addr = ((FixupPrecode*)pPrecode)->GetTargetSlot();
-                }
+                accessType = IAT_PVALUE;
+                targetAddr = ((FixupPrecode*)pPrecode)->GetTargetSlot();
+#endif // FEATURE_PORTABLE_ENTRYPOINTS
                 goto exit;
             }
         }
 
-        pfnHelper = LoadDynamicJitHelper((DynamicCorInfoHelpFunc)dynamicFtnNum).pfnHelper;
+        pfnHelper = LoadDynamicJitHelper((DynamicCorInfoHelpFunc)dynamicFtnNum);
```

Comment:
With portable entrypoints, this returns `PortableEntryPoint*` but then we will go and wrap it in PortableEntryPoint a few lines later again since ` if (targetAddr == NULL)` condition is going to be true. I do not see how this can work.